zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/exception/RestoreException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zookeeper.server.backup.exception;
+
+/**
+ * Signals an out-of-range error has occurred during the ZooKeeper restore process.
+ * For example: User requested to restore to a timestamp earlier than the earliest timestamp recorded in timetable,
```

Comment:
This usage (timestamp being out of range) is currently represented by an IllegalArgumentException in TimetableUtil. I think either is fine, but if we want to introduce another exception, we should update `TimetableUtil` just to keep things consistent. There are other modes of failure in `TimetableUtil` that could potentially use `RestoreException` as well, so if we want to include those failure modes, we should update accordingly in this PR.

I am personally not entirely sure if this extra exception is needed. Perhaps it's a good idea to use this for restoration-related logic (other than IllegalArgument, IOException, etc.), but one thing I would encourage is to change the scope of the out-of-range error because for that particular error case, I think IllegalArgument might be more appropriate? What do you think?